plugin/go.vim

```diff
@@ -130,8 +130,14 @@ function! s:GoInstallBinaries(updateBinaries, ...)
     endfor
   else
     let l:packages = s:packages
+    " Filter packages from exclude list, if no binaries were explicitly
+    " specified.
+    for l:bin in go#config#GoExcludeBinaries()
+      call remove(l:bin, l:packages)
+    endfor
```

Comment:
`l:packages` needs to copy `s:packages` (use the `copy` or `deepcopy`) since entries are being removed so that `s:packages` is not altered.

Also, `:GoInstallBinaries` should probably not skip anything that's missing. I think that means that instead of this removal, the check around line 160 that currently reads
```
      if a:updateBinaries == 1
        echo "vim-go: Updating " . l:binary . ". Reinstalling ". importPath . " to folder " . go_bin_path
```
should probably be something like
```
      if a:updateBinaries == 1
        if index(go#config#GoExcludeBinaries(), l:bin) != -1
            echo "vim-go: Skipping out of date " . l:binary . ". "
            continue
        endif
        echo "vim-go: Updating " . l:binary . ". Reinstalling ". importPath . " to folder " . go_bin_path
```